ConcatenateMatrices.java

```diff
@@ -3,37 +3,70 @@
 
 public class ConcatenateMatrices {
 
-    public static void main(String[] args) throws IOException {
-        Scanner scanner = new Scanner(new File("vstup.txt"));
-        PrintStream output = new PrintStream("vystup.txt");
-        String vstup = scanner.nextLine(); //prvy riadok (velmi pravdepodobne ide nakodit lahsie)
-        String[] rozmery = vstup.split(" ");
-        int m = Integer.parseInt(rozmery[0]);
-        int n = Integer.parseInt(rozmery[1]);
+    public static void main(String[] args){
 
-        String[][] matrix = new String[m][]; //matica a jej vynulovanie
-        for (int k = 0; k < m; k++) {
-            matrix[k] = new String[n];
+        Scanner scanner = null;
+        PrintStream output = null;
+        try {
+            scanner = new Scanner(new File("vstup.txt"));
+            output = new PrintStream("vystup.txt");
         }
+        catch (Exception e){
+            System.out.println("Problem s nacitanim/vytvorenim vstupneho/vystupneho suboru.");
+            if(scanner != null) scanner.close();
+            if(output != null) output.close();
+            return;
+        }
+
+        int m, n;
+        try {
+            m = scanner.nextInt();
+            n = scanner.nextInt();
+            scanner.nextLine();
+            if(m < 1 || n < 1) throw new IllegalArgumentException();
+        }
+        catch (Exception e){
+            System.out.println("Nespravne rozmery matic.");
+            if(scanner != null) scanner.close();
+            if(output != null) output.close();
+            return;
+        }
+
+        String[][] matrix = new String[m][n]; //matica a jej vynulovanie
         for (int i = 0; i < m; i++) {
             for (int j = 0; j < n; j++) {
                 matrix[i][j] = "";
             }
         }
+
         while (scanner.hasNextLine()) { //cita pokial nenarazi na koniec (neexistujuci prvy riadok matice)
-            for (int i = 0; i < m; i++) { //ak je dalsi riadok, tak urcite bude m-riadkov matice
+            for (int i = 0; i < m; i++) {
+                if(!scanner.hasNextLine()){
+                    System.out.println("Nedostatocny pocet zadanych riadkov matice.");
+                    if(scanner != null) scanner.close();
+                    if(output != null) output.close();
```

Comment:
Doing the same test several times is definitely not the way to go. Such corner cases should be dealt with upfront to avoid polluting the code. Read about happy paths:
https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88